In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [18]:
#Reading the dataset
dataset = pd.read_csv("training_data.csv")

In [27]:
dataset.head()

CHELSA_bio10_01_b1  FAPAR_mean_January  FAPAR_mean_June  precipitation  \
0                 193          621.600000       628.000000    15008.91667   
1                 173          385.458333       556.000000    15157.66667   
3                 295          168.326531       212.294118    15771.58333   
4                 220          438.111111       587.333333    15225.25000   
5                 277          113.838710       119.661017    15677.08333   

   elevation  landcover_cropland_precentage  landcover_class  landsat8_nir  \
0       1681                             11                4            65   
1       2201                             52                4            67   
3        387                              0                3            49   
4       1114                             27                4            66   
5        749                              0                7            61   

   landsat8_red  landsat8_swir1  landsat8_swir2   pH  sentinel2_band_2  \
0            20              62              39  6.4                28   
1            45             105              81  6.8                31   
3            59              72              68  8.1                23   
4            29              67              38  7.5                15   
5            94             105             108  7.7                57   

   sentinel2_band_4  sentinel2_band_9  
0                37               123  
1                45                96  
3                41       10000000000  
4                18               130  
5                95               121

In [21]:
len(dataset)

8932

Dropping NaN values

In [22]:
dataset = dataset.dropna()

In [23]:
len(dataset)

8932

In [24]:
dataset.head()

CHELSA_bio10_01_b1  FAPAR_mean_January  FAPAR_mean_June  precipitation  \
0                 193          621.600000       628.000000    15008.91667   
1                 173          385.458333       556.000000    15157.66667   
3                 295          168.326531       212.294118    15771.58333   
4                 220          438.111111       587.333333    15225.25000   
5                 277          113.838710       119.661017    15677.08333   

   elevation  landcover_cropland_precentage  landcover_class  landsat8_nir  \
0       1681                             11                4            65   
1       2201                             52                4            67   
3        387                              0                3            49   
4       1114                             27                4            66   
5        749                              0                7            61   

   landsat8_red  landsat8_swir1  landsat8_swir2   pH  sentinel2_band_2  \
0            20              62              39  6.4                28   
1            45             105              81  6.8                31   
3            59              72              68  8.1                23   
4            29              67              38  7.5                15   
5            94             105             108  7.7                57   

   sentinel2_band_4  sentinel2_band_9  
0                37               123  
1                45                96  
3                41       10000000000  
4                18               130  
5                95               121

In [25]:
dataset.columns

Index(['CHELSA_bio10_01_b1', 'FAPAR_mean_January', 'FAPAR_mean_June',
       'precipitation', 'elevation', 'landcover_cropland_precentage',
       'landcover_class', 'landsat8_nir', 'landsat8_red', 'landsat8_swir1',
       'landsat8_swir2', 'pH', 'sentinel2_band_2', 'sentinel2_band_4',
       'sentinel2_band_9'],
      dtype='object')

In [28]:
#Setting the value for X and Y

x = dataset[['landsat8_nir', 'landsat8_red', 'landsat8_swir1',
       'landsat8_swir2', 'sentinel2_band_2', 'sentinel2_band_4',
       'sentinel2_band_9']]
y = dataset['pH']

Splitting the dataset into train and test set

In [29]:
from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.3, random_state = 100)

Implementing the linear model

In [30]:
from sklearn.linear_model import LinearRegression

In [31]:
#Fitting the Multiple Linear Regression model

LR = LinearRegression()

# fitting the training data

LR.fit(x_train,y_train)

LinearRegression()

Prediction of pH

In [32]:
y_prediction =  LR.predict(x_test)
y_prediction

array([6.78603394, 6.22394711, 5.92298656, ..., 5.82598075, 6.3888107 ,
       6.38251337])

Model Equation

In [33]:
print("Intercept: ", LR.intercept_)
print("Coefficients:")
list(zip(x, LR.coef_))

Intercept:  7.411516807276677
Coefficients:


[('landsat8_nir', -0.008226991549768643),
 ('landsat8_red', -0.0015591382320334667),
 ('landsat8_swir1', -0.034745386283111045),
 ('landsat8_swir2', 0.0392591495289327),
 ('sentinel2_band_2', 0.04517669967014346),
 ('sentinel2_band_4', -0.021456796845489727),
 ('sentinel2_band_9', 8.56303803906111e-11)]

Regression Equation: pH = 7.4115 + (-0.008227 * landsat8_nir) + (-0.001559 * landsat8_red) + (landsat8_swir1 * -0.03475) + (landsat8_swir2 * 0.03925) + (sentinel2_band_2 * 0.04518) + (sentinel2_band_4 * -0.021456) + (sentinel2_band_9 * 8.5630e-11)

Prediction on the test set

In [34]:
#Prediction of test set

y_pred_mlr= LR.predict(x_test)

Predicted values

In [35]:
print("Prediction for test set: {}".format(y_pred_mlr))

Prediction for test set: [6.78603394 6.22394711 5.92298656 ... 5.82598075 6.3888107  6.38251337]


Actual values and the predicted values

In [36]:
mlr_diff = pd.DataFrame({'Actual value': y_test, 'Predicted value': y_pred_mlr})
mlr_diff.head()

Actual value  Predicted value
7844          7.2         6.786034
834             6         6.223947
2507          4.6         5.922987
2671            6         6.095407
7232          5.8         5.960830